cpp/src/ray/config_internal.cc

```diff
@@ -46,8 +46,24 @@ void ConfigInternal::Init(RayConfig &config, int *argc, char ***argv) {
     // Parse config from command line.
     gflags::ParseCommandLineFlags(argc, argv, true);
 
-    if (!FLAGS_ray_dynamic_library_path.empty()) {
-      dynamic_library_path = FLAGS_ray_dynamic_library_path;
+    if (!FLAGS_ray_code_search_path.empty()) {
+      // Code search path like this "/path1/xxx.so:/path2".
+      code_search_path.clear();
+      std::string separator = ":";
+      unsigned int start = 0;
+      auto end = FLAGS_ray_code_search_path.find(separator);
+      if (end == std::string::npos) {
+        // A single path without separator.
+        code_search_path.emplace_back(FLAGS_ray_code_search_path);
+      } else {
+        // Multiple path with separator.
+        while (end != std::string::npos) {
+          code_search_path.emplace_back(
+              FLAGS_ray_code_search_path.substr(start, end - start));
+          start = end + separator.length();
+          end = FLAGS_ray_code_search_path.find(separator, start);
```

Comment:
You missed the last segment if there are multiple paths. Have you tested multiple paths?

```suggestion
      while (start < FLAGS_ray_code_search_path.size()) {
        auto end = FLAGS_ray_code_search_path.find(separator, start);
        if (end == std::string::npos) {
          end = FLAGS_ray_code_search_path.size();
        }
        if (start < end) { // In case there are unnecessary separators.
          code_search_path.emplace_back(FLAGS_ray_code_search_path.substr(start, end - start);
        }
        start = end + 1;
      }
```

Reply:
Good~